core/internal/stream/stream.go

```diff
@@ -445,6 +473,11 @@
 	s.runWork.Close()
 	s.wg.Wait()
 	s.logger.Info("stream: closed", "id", s.settings.GetRunID())
+
+	if s.loggerFile != nil {
+		// Sync the file instead of closing it, in case we keep writing to it.
+		_ = s.loggerFile.Sync()
```

Comment:
how about adding a timeout? if it's say a slow nfs or something, this could potentially block for a long time

Reply:
It doesn't block the Python process except on teardown (when the Python script is done and waiting to exit), in which case blocking until a clean shutdown is good. Note that the `Close()` for `debug-core.log` and the Writer will block in a similar way.